src/streamlines.py

```diff
@@ -0,0 +1,246 @@
+        row=starting_row + 7, sticky=Tk.E
+    )
+
+
+def __show_streamline_handler(settings, panel):
+    """Handle what happens when the `show_streamlines` button is toggledself.
+
+    Parameters
+    ----------
+    settings :
+        The settings object for the panel being drawn in
+    panel :
+        The Panel object the streamlines are being drawn in
+    """
+    # Write value to the settings dictionary
+    settings.parent.SetPlotParam(
+        "show_streamlines", settings.show_streamlines.get(), update_plot=False
+    )
+    # Either create the streamlines or remove them depending on the state of the checkbox
+    if settings.parent.GetPlotParam("show_streamlines"):
+        draw_streamlines(panel)
+    else:
+        remove_streamlines(panel)
+
+    # Update everything
+    settings.parent.parent.canvas.draw()
+    settings.parent.parent.canvas.get_tk_widget().update_idletasks()
+
+
+def __update_parameter(settings, param_name, value):
+    settings.parent.SetPlotParam(param_name, value, update_plot=True)
+
+
+def streamlines_callback(settings, update_plot=True):
+    # Don't update the plot if there's no streamplot active
+    if not settings.parent.GetPlotParam("show_streamlines"):
+        update_plot = False
+
+    # Handle streamline stride
+    if settings.streamlines_stride.get() != settings.streamlines_stride:
+        settings.parent.SetPlotParam(
+            "streamlines_stride",
+            settings.streamlines_stride.get(),
+            update_plot=update_plot,
+        )
+
+    # Handle streamline density
+    if settings.streamlines_density.get() != settings.streamlines_density:
+        settings.parent.SetPlotParam(
+            "streamlines_density",
+            settings.streamlines_density.get(),
+            update_plot=update_plot,
+        )
+
+    # Handle streamline color
+    if settings.streamlines_color.get() != settings.streamlines_color:
+        settings.parent.SetPlotParam(
+            "streamlines_color",
+            settings.streamlines_color.get(),
+            update_plot=update_plot,
+        )
+
+
+def draw_streamlines(panel):
+    """Draw streamlines.
+
+    Parameters
+    ----------
+    panel :
+        The Panel object the streamlines are being drawn in
+    """
+    # Get settings from panel
+    fields = panel.GetPlotParam("streamlines_fields")
+    stride = panel.GetPlotParam("streamlines_stride")
+    line_density = panel.GetPlotParam("streamlines_density")
+    line_color = panel.GetPlotParam("streamlines_color")
+
+    # Grab the data, use aliases to make it easier to work with
+    if fields == "bxby":
+        bx_name, by_name = "bx", "by"
+    elif fields == "bybz":
+        bx_name, by_name = "by", "bz"
+    elif fields == "bxbz":
+        bx_name, by_name = "bx", "bz"
+    else:
+        raise ValueError(
+            f"Improper fields value of {fields} passed to draw_streamlines. "
+        )
+    bx = panel.parent.DataDict[bx_name][0, ::stride, ::stride]
```

Comment:
In this slice and the next you need to make sure you are on the Z-index when you are in the x-y plane not defaulting to the zero index.

Similar if you are in the x-z plane you need to have the right Y-index, and taking the correct slice, so the indexes you'll stride in will change. See the fields plot for an example.

Otherwise great work!

Reply:
Just to clarify, Iseult uses numpy's column major ordering (`[x,y,z]`) not Fortran's row major ordering right? (`[z,y,x]`). I now Tristan uses the latter but numpy defaults to the former.

Reply:
Thanks for pointing this out, I believe it's fixed with the latest commit.